src/lib.rs

```diff
@@ -170,8 +271,13 @@ pub mod __rt {
             if std::thread::panicking() {
                 return;
             }
-            let value_on_exit = self.mark.load(Ordering::Relaxed);
-            assert!(value_on_exit > self.value_on_entry, "mark was not hit")
+            let value_on_exit = self.mark.value();
+            if let Some(hits) = self.expected_hits {
+                let hit_count = value_on_exit.wrapping_sub(self.value_on_entry);
+                assert!(hit_count == hits, "mark was hit {} times, expected {}", hit_count, hits)
+            } else {
+                assert!(value_on_exit > self.value_on_entry, "mark was not hit")
+            }
```

Comment:
I have a small idiosyncrastic preference towards `match` over `if let else`, but this is also fine.

Reply:
it is match now!